src/server/game/Handlers/GroupHandler.cpp

```diff
@@ -1162,121 +737,110 @@ void WorldSession::HandleRequestPartyMemberStatsOpcode(WorldPacket& recvData)
 
-    data.put<uint64>(maskPos, auramask);                    // GROUP_UPDATE_FLAG_AURAS
-
-    if (updateFlags & GROUP_UPDATE_FLAG_PET_GUID)
-        data << pet->GetGUID();
-
-    data << std::string(pet ? pet->GetName() : "");         // GROUP_UPDATE_FLAG_PET_NAME
-    data << uint16(pet ? pet->GetDisplayId() : 0);          // GROUP_UPDATE_FLAG_PET_MODEL_ID
-
-    if (updateFlags & GROUP_UPDATE_FLAG_PET_CUR_HP)
-        data << uint32(pet->GetHealth());
-
-    if (updateFlags & GROUP_UPDATE_FLAG_PET_MAX_HP)
-        data << uint32(pet->GetMaxHealth());
-
-    if (updateFlags & GROUP_UPDATE_FLAG_PET_POWER_TYPE)
-        data << (uint8)pet->getPowerType();
-
-    if (updateFlags & GROUP_UPDATE_FLAG_PET_CUR_POWER)
-        data << uint16(pet->GetPower(pet->getPowerType()));
-
-    if (updateFlags & GROUP_UPDATE_FLAG_PET_MAX_POWER)
-        data << uint16(pet->GetMaxPower(pet->getPowerType()));
-
-    // GROUP_UPDATE_FLAG_PET_AURAS
-    uint64 petAuraMask = 0;
-    data << uint8(1);
-    maskPos = data.wpos();
-    data << uint64(petAuraMask);                            // placeholder
-    data << uint32(MAX_AURAS);                              // count
-    if (pet)
+    partyMemberState.HasPet = pet ? true : false;
+    if (partyMemberState.HasPet)
     {
+        partyMemberState.PetName = pet->GetName();                          // GROUP_UPDATE_FLAG_PET_NAME
+        partyMemberState.PetModelId = uint16(pet->GetDisplayId());          // GROUP_UPDATE_FLAG_PET_MODEL_ID
+        partyMemberState.PetCurrentHealth = uint32(pet->GetHealth());
+        partyMemberState.PetMaxHealth = uint32(pet->GetMaxHealth());
+
+        uint32 petAuraCounter = 0;
         for (uint8 i = 0; i < MAX_AURAS; ++i)
         {
             if (AuraApplication const* aurApp = pet->GetVisibleAura(i))
             {
-                petAuraMask |= (uint64(1) << i);
-
-                data << uint32(aurApp->GetBase()->GetId());
-                data << uint16(aurApp->GetFlags());
+                WorldPackets::Party::PartyMemberState::Aura aura;
+                aura.SpellId = uint32(aurApp->GetBase()->GetId());
+                //todo: Scalings
+                aura.EffectMask = uint32(aurApp->GetEffectMask());
 
                 if (aurApp->GetFlags() & AFLAG_SCALABLE)
                 {
                     for (uint32 i = 0; i < MAX_SPELL_EFFECTS; ++i)
                     {
                         if (AuraEffect const* eff = aurApp->GetBase()->GetEffect(i))
-                            data << int32(eff->GetAmount());
+                            aura.Scales.push_back(float(eff->GetAmount()));
                         else
-                            data << int32(0);
+                            aura.Scales.push_back(float(0));
                     }
                 }
+                petAuraCounter++;
+                partyMemberState.PetAuraList.push_back(aura);
             }
         }
+        partyMemberState.PetAuraCount = petAuraCounter;
```

Comment:
this is not needed at all. when writing length to the packet write `packet << uint32(partyMemberState.PetAuraList.size());`